src/components/Authentication/Register.vue

```diff
@@ -0,0 +1,115 @@
+<template lang="pug">
+.register-widget
+  v-alert.mt-0(
+      dismissible,
+      transition="scale-transition",
+      v-for="err in alerts.errors",
+      :key="err",
+      :value="true",
+      type="error") {{ err }}
+  v-container
+    v-form(@submit.prevent="register", ref="register")
+      v-text-field(
+          v-model="login",
+          label="Username",
+          :rules="nonEmptyRules",
+          type="text",
+          autofocus)
+      v-text-field(
+          v-model="email",
+          label="Email",
+          :rules="nonEmptyRules",
+          type="email")
+      v-text-field(
+          v-model="firstName",
+          label="First Name",
+          :rules="nonEmptyRules",
+          type="text")
+      v-text-field(
+          v-model="lastName",
+          label="Last Name",
+          :rules="nonEmptyRules",
+          type="text")
+      v-text-field(
+          v-model="password",
+          type="password",
+          label="Password",
+          :rules="nonEmptyRules")
+      v-text-field(
+          v-model="retypePassword",
+          type="password",
+          label="Retype password",
+          :rules="retypeMustMatchPasswordRules")
+      v-btn.ml-0(type="submit",
+          color="primary",
+          :loading="inProgress") Register
+  v-divider(v-if="oauthProviders.length")
+  oauth(v-if="oauthProviders.length",
+      verb="register",
+      :providers="oauthProviders")
+</template>
+
+<script>
+import Oauth from './OAuth.vue';
+
+export default {
+  inject: ['girderRest'],
+  props: {
+    oauthProviders: {
+      type: Array,
+      default: () => [],
+    },
+  },
+  components: {
+    Oauth,
+  },
+  data() {
+    const nonEmptyRules = [v => !!v || 'Item is required'];
+    const retypeMustMatchPasswordRules = nonEmptyRules.concat([
+      v => v === this.password || 'Passwords must match',
+    ]);
+    return {
+      login: '',
+      email: '',
+      firstName: '',
+      lastName: '',
+      password: '',
+      retypePassword: '',
+      retypePasswordErrors: [],
+      inProgress: false,
+      alerts: {
+        errors: [],
+      },
+      nonEmptyRules,
+      retypeMustMatchPasswordRules,
+    };
+  },
+  methods: {
+    async register() {
+      if (!this.$refs.register.validate()) {
+        return;
+      }
+      this.inProgress = true;
+      this.alerts.errors.length = 0;
```

Comment:
I didn't realize this actually works as a way to empty an Array, you taught me something new! (I would've expected `length` to be a read only property.)

To follow the Principle of Least Astonishment, it's probably best to instead do `this.alerts.errors = [];`.